### PR TITLE
Typedoc setup

### PR DESCRIPTION
**Pull request by rvdkooy**
Hi,

I took the opportunity to look into some generated docs based on the sourcecode
Typedoc seems pretty neat: https://typedoc.org/

Is this something you want to look into further regarding https://github.com/pamapa/oidc-client-ts/issues/15?

some screenshots:

image

image

R

### CODE REVIEW

**Comment by pamapa** [comment]
You need to remove locally first the package-lock.json and then npm install again, to have it clean.

File: src/UserManager.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ type ExtraSigninRequestArgs = Pick<CreateSigninRequestArgs, "extraQueryParams" |
 type ExtraSignoutRequestArgs = Pick<CreateSignoutRequestArgs, "extraQueryParams">
 
 /**
- * @public
```
Comment (pamapa) [inline src/UserManager.ts:24]: The @public is needed for the API extractor...

**Comment by pamapa** [comment]
The main question is if we should add typedoc at all, now that we have the API extractor with its generated docs/oidc-client-ts.api.md file. What do the others mean? @kherock  @PascalLuginbuehl

**Comment by kherock** [comment]
I think the typedoc is a lot more navigable and has value on its own. The API report is nice if you want to see the git history of the API within thie repostiory. I don't think the report markdown is particularly useful on the website when tools like typedoc do a better job.

**Comment by pamapa** [comment]
You will need to rebase against remote/main + also squash your commits into one.

pipeline warning/error:
```
Warning: You have changed the public API signature for this project. Please copy the file "temp/oidc-client-ts.api.md" to "docs/oidc-client-ts.api.md", or perform a local build (which does this automatically). See the Git repo documentation for more info.
```

**Comment by pamapa** [comment]
superseded by #189

**Pull request closed without merging**